lib/internal/errors.js

```diff
@@ -0,0 +1,258 @@
+'use strict';
+
+// The whole point behind this internal module is to allow Node.js to no
+// longer be forced to treat every error message change as a semver-major
+// change. The NodeError classes here all expose a `code` property whose
+// value identifies the error.
+
+const kCode = Symbol('code');
+
+class NodeError extends Error {
+  constructor(key, ...args) {
+    super(message(key, args));
+    this[kCode] = key;
+    Error.captureStackTrace(this, NodeError);
```

Comment:
Not that it matters to me, but this means it'll be taking 2 stack traces for every error and will be a performance hit. Just saying, someone may make a stink about out.